KEYCLOAK-16679 Add algorithm settings for client assertion signature in OIDC identity broker

i7a7467 opened the pull request:
https://issues.redhat.com/browse/KEYCLOAK-16679

<!---
Please read https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

i7a7467 committed:
KEYCLOAK-16679 Add algorithm settings for client assertion signature in OIDC identity broker

mposolda commented on themes/src/main/resources/theme/base/admin/messages/admin-messages_en.properties:
```diff
@@ -644,6 +644,8 @@ client-auth.client_secret_jwt=Client secret as jwt
 identity-provider.client-id.tooltip=The client or client identifier registered within the identity provider.
 client-secret=Client Secret
+client-assertion-signing-algorithm=Client Assertion Signature Algorithm
+client-assertion-signing-algorithm.tooltip=JWA Algorithm used for client assertions.
```
Is it possible to add slightly more details to this tooltip to specify what exactly does this option mean, when it is needed to be configured etc? My vote is just include everything, which you added to the documentation to this tooltip. So something like this: `Signature algorithm to create JWT assertion as client authentication. In the case of JWT signed with private key or Client secret as jwt, it is required. If no algorithm is specified, the following algorithm is adapted. `RS256` is adapted the in the case of JWT signed with private key.  `HS256` is adapted the in the case of Client secret as jwt.`

i7a7467 replied:
That's exactly what you say. 
I added the same content as the documentation to tooltip message.

mposolda reviewed (changes requested):
@i7a7467 Thanks for the PR and sorry for the late review. Few comments:
- Rebase will be needed
- I suggested inline to slightly change the tooltip to make it more clear what does this means
- Automated test will be needed

mposolda commented:
Docs PR: https://github.com/keycloak/keycloak-documentation/pull/1075

i7a7467 commented:
@mposolda 
Thank you for your review comment.
I fixed the review points. I rebased.

### About automated tests
I added tests using private_key_jwt and client_secret_jwt authentication methods and non-default algorithms.
- private_key_jwt:ES256
- client_secret_jwt:HS384

mposolda reviewed (approved):
@i7a7467 Thanks!
